libexec/scoop-search.ps1

```diff
@@ -37,19 +37,30 @@ function bin_match($manifest, $query) {
 }
 
 function search_bucket($bucket, $query) {
-    $apps = apps_in_bucket (Find-BucketDirectory $bucket) | ForEach-Object { @{ name = $_ } }
-
-    if ($query) {
-        $apps = $apps | Where-Object {
-            if ($_.name -match $query) { return $true }
-            $bin = bin_match (manifest $_.name $bucket) $query
+    $apps = Get-ChildItem (Find-BucketDirectory $bucket) -Filter '*.json' -Recurse
+
+    $apps | ForEach-Object {
+        $json = [System.IO.File]::ReadAllText($_.FullName) | ConvertFrom-Json -ErrorAction Continue
+
+        if($app.name -match $query) {
+            $list.Add([PSCustomObject]@{
+                Name = $_.BaseName
```

Comment:
```suggestion
        $json = [System.IO.File]::ReadAllText($_.FullName) | ConvertFrom-Json -ErrorAction Continue
        $name = $_.BaseName

        if ($name -match $query) {
            $list.Add([PSCustomObject]@{
                Name = $name
```

Reply:
We don't have a variable called `$app`, so all search fallback to bin search.

![image](https://user-images.githubusercontent.com/5832170/211159247-867c76d4-6057-4eab-849b-5a9d151c1f2b.png)

Reply:
yep, missed that one :smile: